app/services/course_creation_step_service.rb

```diff
@@ -0,0 +1,139 @@
+class CourseCreationStepService
+  def execute(current_step:, course:)
+    workflow_steps = get_workflow_steps(course)
+
+    {
+      next: get_next_step(workflow_steps, current_step),
+      previous: get_previous_step(workflow_steps, current_step),
+    }
+  end
+
+  def get_next_step(steps, current_step)
+    steps[steps.find_index(current_step).next]
+  end
+
+  def get_previous_step(steps, current_step)
+    steps[steps.find_index(current_step).pred]
+  end
+
+  def get_workflow_steps(course)
+    if course.is_further_education?
+      further_education_workflow_steps(course)
+    elsif course.is_uni_or_scitt?
+      uni_or_scitt_workflow_steps(course)
+    elsif course.is_school_direct?
+      school_direct_workflow_steps(course)
+    end
+  end
+
+  def school_direct_workflow_steps(course)
```

Comment:
So - I'm gonna compare this with how we've managed stuff in #791. Are we gonna do a similar thing with modern languages? 

Feels weird that one has the controller decide, and one has here decide. I think it probably _should_ be handled in here though :thinking:

Reply:
The controllers should decide whether or not to skip a step in a given workflow, I've now changed things to reflect that